utils/dependency_management.py

```diff
@@ -161,37 +161,43 @@ def generate_environment_yml():
 
 
 @cli.command()
-def generate_pyproject_toml():
+def update_pyproject_toml():
     """Generate 'pyproject.toml' file."""
 
+    # read the current file
+    toml_path = ROOT / 'pyproject.toml'
+    if toml_path.exists():
+        pyproject = toml.loads(toml_path.read_text(encoding='utf8'))
+    else:
+        pyproject = {}
+
     # Read the requirements from 'setup.json'
     setup_cfg = _load_setup_cfg()
     install_requirements = [Requirement.parse(r) for r in setup_cfg['install_requires']]
-
     for requirement in install_requirements:
         if requirement.name == 'reentry':
             reentry_requirement = requirement
             break
     else:
         raise DependencySpecificationError("Failed to find reentry requirement in 'setup.json'.")
 
-    pyproject = {
-        'build-system': {
-            'requires': ['setuptools>=40.8.0,<50', 'wheel',
-                         str(reentry_requirement), 'fastentrypoints~=0.12'],
-            'build-backend': 'setuptools.build_meta:__legacy__',
-        }
+    # update the build-system key
+    pyproject['build-system'] = {
+        'requires': ['setuptools>=40.8.0,<50', 'wheel',
+                     str(reentry_requirement), 'fastentrypoints~=0.12'],
+        'build-backend': 'setuptools.build_meta:__legacy__',
     }
```

Comment:
```suggestion
    pyproject.setdefault('build-system', {})
    pyproject['build-system'].update({
        'requires': ['setuptools>=40.8.0,<50', 'wheel',
                     str(reentry_requirement), 'fastentrypoints~=0.12'],
        'build-backend': 'setuptools.build_meta:__legacy__',
    })
```

is this what you had in mind?

Reply:
Yes. You may even be able to chain it, but I'm not 100% sure on copy-vs-reference behavior here.

Reply:
Yeh I couldn't remember off hand either, so I thought better safe than sorry lol

Reply:
You can chain it, but this is more readable IMO.